!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

_This issue was originally opened by @himi64 as hashicorp/terraform#18523. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
```
Terraform v0.11.7
+ provider.aws v1.27.0
```

### Terraform Configuration Files
```
### ECR ########################################################################
resource "aws_ecr_repository" "repo1" {
  name = "${var.ECR_NAME}"
}

### GET ROLE DATA
data "aws_iam_role" "ecsInstanceRole" {
  name = "ecsInstanceRole"
}
data "aws_iam_role" "aws-ec2-spot-fleet-tagging-role" {
  name = "aws-ec2-spot-fleet-tagging-role"
}
data "aws_iam_role" "AWSBatchServiceRole" {
  name = "AWSBatchServiceRole"
}

### BATCH ######################################################################
### COMPUTE ENVIRONMENT
resource "aws_batch_compute_environment" "TEST-ENV" {
  compute_environment_name = "TEST-ENV"
  compute_resources {
    instance_role = "${aws_iam_role.AWSBatchServiceRole.id}"
    instance_type = [
      "c4.large",
    ]
    spot_iam_fleet_role = "${aws_iam_role.aws-ec2-spot-fleet-tagging-role.id}"
    bid_percentage = 0.1
    max_vcpus = 8
    min_vcpus = 0
    security_group_ids = [
      "${var.security_group_id}"
    ]
#    "${var.SECURITY_GROUP_ID}"
    subnets = [
      "${var.subnet_id}"
    ]
    type = "SPOT"
  }
  service_role = "${aws_iam_role.ecsInstanceRole.id}"
  type = "MANAGED"
}

### JOB QUEUE
resource "aws_batch_job_queue" "test-queue" {
  name = "test-queue"
  state = "ENABLED"
  priority = 1
  compute_environments = ["${aws_batch_compute_environment.TEST-ENV.arn}"]
}

### JOB DEFINITION
resource "aws_batch_job_definition" "test-job-def" {
    name = "${var.ECR_NAME}"
    type = "container"
}

### CLOUDWATCH #################################################################
### ROLE FOR BATCH TO SUBMIT JOB
resource "aws_iam_role" "batch_test_role" {
  name = "test_role"
  assume_role_policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "batch.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
EOF
}

# Attach the above role to a policy
resource "aws_iam_role_policy" "test_policy" {
  name = "test_policy"
  role = "${aws_iam_role.batch_test_role.id}"
  policy = <<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "batch:SubmitJob"
            ],
            "Resource": "*"
        }
    ]
}
EOF
}

### 2. EVENT TARGETS
resource "aws_cloudwatch_event_target" "scheduled-job" {
  target_id = "scheduled-job"
  rule      = "${aws_cloudwatch_event_rule.scheduled-job.name}"
  arn  = "${aws_batch_job_queue.test-queue.arn}"
  role_arn = "${aws_iam_role.batch_test_role.arn}" # "arn:aws:iam::984374187558:role/service-role/AWS_Events_Invoke_Batch_Job_Queue_253875752"
  batch_target {
    job_name = "test"
    job_definition = "${aws_batch_job_definition.test-job-def.id}" #try .arn
  }
}

### 3. EVENT RULES
resource "aws_cloudwatch_event_rule" "scheduled-job" {
  name                = "scheduled-job"
  description         = "Cron Job"
  schedule_expression = "cron(30 22 ? * * *)"
}
```
### Debug Output
https://gist.github.com/himi64/d2129615cc8c5be18d00e757dfb97c35

### Crash Output
https://gist.github.com/himi64/c781a7733e423ab34bbd58c37ed79cab

### Expected Behavior
Expected Terraform to apply the defined resources to the environment.

### Actual Behavior
Terraform Crashed.

### Steps to Reproduce
1. Define the environment resources in the .tf script
2. `terraform init`
3. `terraform apply`

### Additional Context
This script used to work fine until last Friday, July 20th 2018 (first time terraform crashed).

### References

Hi @himi64 👋  Thanks for reporting this and sorry you ran into this unexpected issue.

In the crash log it appears the panic is occurring due to this:

```
panic: Error reading level config: strconv.ParseInt: parsing "0.1": invalid syntax
```

Which unhelpfully is likely caused by this in your configuration:

```
bid_percentage = 0.1
```

Does this work if you change your configuration to the below?

```hcl
bid_percentage = 10
```

The `bid_percentage` attribute is currently `schema.TypeInt` ([code reference](https://github.com/terraform-providers/terraform-provider-aws/blob/e61ed21f133d387e10d0df683a09f193d6cf0df3/aws/resource_aws_batch_compute_environment.go#L37)) which seems like its not properly handling a float value. While we can fix the code to return a type conversion error more appropriately in this situation, this change is global across all providers and will need to occur upstream in Terraform core.

In the meantime, I think its probably appropriate we mark this provider issue to adjust our [`bid_percentage` attribute documentation](https://www.terraform.io/docs/providers/aws/r/batch_compute_environment.html#bid_percentage) to reflect that an integer is expected here rather than a float value so I will tag this issue in that regard.

Please let us know if the above configuration change fixes your issue. Thanks!

--- a/website/docs/r/batch_compute_environment.html.markdown
+++ b/website/docs/r/batch_compute_environment.html.markdown
@@ -116,7 +116,7 @@ resource "aws_batch_compute_environment" "sample" {
 
 **compute_resources** is a child block with a single argument:
 
-* `bid_percentage` - (Optional) The minimum percentage that a Spot Instance price must be when compared with the On-Demand price for that instance type before instances are launched. For example, if your bid percentage is 20%, then the Spot price must be below 20% of the current On-Demand price for that EC2 instance. This parameter is required for SPOT compute environments.
+* `bid_percentage` - (Optional) Integer of minimum percentage that a Spot Instance price must be when compared with the On-Demand price for that instance type before instances are launched. For example, if your bid percentage is 20% (`20`), then the Spot price must be below 20% of the current On-Demand price for that EC2 instance. This parameter is required for SPOT compute environments.
 * `desired_vcpus` - (Optional) The desired number of EC2 vCPUS in the compute environment.
 * `ec2_key_pair` - (Optional) The EC2 key pair that is used for instances launched in the compute environment.
 * `image_id` - (Optional) The Amazon Machine Image (AMI) ID used for instances launched in the compute environment.

